src/context/TotalInOutBalanceChartContext/TotalInOutBalanceChartContextProvider.tsx

```diff
@@ -62,13 +65,27 @@ const TotalInOutBalanceChartContextProvider: FC<PropsWithChildren> = ({
     memoizedQueryVariables.queryOptions,
   );
 
+  const cancelQuery = useCallback(() => {
+    if (loading) {
+      memoizedQueryVariables.abortController.abort();
+    }
+  }, [memoizedQueryVariables.abortController, loading]);
+
   useEffect(() => {
```

Comment:
do we still need both useEffects since the hook for the funding cards only does it in one `useEffect`? Maybe we could try to just have one and see if it works :crossed_fingers: